src/pl/senla/hotel/ie/ReaderOrder.java

```diff
@@ -0,0 +1,54 @@
+package pl.senla.hotel.ie;
+
+import com.opencsv.CSVParserBuilder;
+import com.opencsv.CSVReader;
+import com.opencsv.CSVReaderBuilder;
+import com.opencsv.ICSVParser;
+import com.opencsv.bean.ColumnPositionMappingStrategy;
+import com.opencsv.bean.CsvToBean;
+import com.opencsv.bean.MappingStrategy;
+import pl.senla.hotel.entity.CsvOrder;
+import pl.senla.hotel.entity.Order;
+import pl.senla.hotel.utils.OrderUtil;
+
+import java.io.FileReader;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.List;
+
+public class ReaderOrder implements Reader<Order> {
```

Comment:
I agree the logic in readers and writers are similar but not the same in all of them.
They use diffenrent Mapping strategies, some of them use additional converters and switchs, one of them use BufferedReader instead of CSVReader...
Could I keep 4 different readers and writers?

Reply:
also, you can merge reader and writer in one class